lightning/src/routing/router.rs

```diff
@@ -4894,6 +4920,32 @@ mod tests {
 			assert_eq!(route.paths[1][0].short_channel_id, 2);
 			assert_eq!(route.paths[1][0].fee_msat, 50_000);
 		}
+
+		{
+			// If we have a bunch of outbound channels to the same node, where most are not
+			// sufficient to pay the full payment, but one is, we should default to just using the
+			// one single channel that has sufficient balance, avoiding MPP.
```

Comment:
If I comment out the sorting, MPP is not used. Could the test be devised such that it would fail with multiple paths as described here when there is no sorting?